### PR TITLE
add cifar-10.1 to tfds

### PR DESCRIPTION
**Pull request by ChanchalKumarMaji**
Related Issue - #738 .

### CODE REVIEW

**Comment by ChanchalKumarMaji** [comment]
While generating url_checksum files, I get ERROR - 
```
0 examples [00:00, ? examples/s]Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/scripts/download_and_prepare.py", line 210, in <module>
    app.run(main)
  File "/home/chanchal/dev_env/lib/python3.6/site-packages/absl/app.py", line 300, in run
    _run_main(main, args)
  File "/home/chanchal/dev_env/lib/python3.6/site-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/scripts/download_and_prepare.py", line 201, in main
    download_and_prepare(builder_for_config)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/scripts/download_and_prepare.py", line 135, in download_and_prepare
    download_config=dl_config,
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/api_utils.py", line 52, in disallow_positional_args_dec
    return fn(*args, **kwargs)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 287, in download_and_prepare
    download_config=download_config)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 948, in _download_and_prepare
    max_examples_per_split=download_config.max_examples_per_split,
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 816, in _download_and_prepare
    self._prepare_split(split_generator, **prepare_split_kwargs)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 965, in _prepare_split
    return self._prepare_split_legacy(generator, split_info)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 955, in _prepare_split_legacy
    self._file_format_adapter.write_from_generator(generator, output_files)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/file_format_adapter.py", line 126, in write_from_generator
    _write_tfrecords_from_generator(wrapped, output_files, shuffle=True)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/file_format_adapter.py", line 234, in _write_tfrecords_from_generator
    _round_robin_write(writers, generator)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/file_format_adapter.py", line 249, in _round_robin_write
    generator, unit=" examples", leave=False)):
  File "/home/chanchal/dev_env/lib/python3.6/site-packages/tqdm/_tqdm.py", line 1017, in __iter__
    for obj in iterable:
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/file_format_adapter.py", line 124, in <genexpr>
    wrapped = (self._serializer.serialize_example(example)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 953, in <genexpr>
    generator = (self.info.features.encode_example(ex) for ex in generator)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/features/features_dict.py", line 168, in encode_example
    in utils.zip_dict(self._feature_dict, example_dict)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/features/features_dict.py", line 167, in <dictcomp>
    for k, (feature, example_value)
  File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/features/class_label_feature.py", line 143, in encode_example
    if not -1 <= example_data < self._num_classes:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

@Conchylicultor @us

File: tensorflow_datasets/image/cifar10_1.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+              "1.0.0", experiments={tfds.core.Experiment.S3: False}),
+          data="v4",
+      ),
+      Cifar10_1Config(
+          name="v6",
+          description=(
+              "It is derived from a slightly improved keyword allocation that is exactly class balanced. This version "
+              "of the dataset corresponds to the results in Appendix D of our paper. v6 contains 2,000 images."),
+          version=tfds.core.Version(
+              "1.0.0", experiments={tfds.core.Experiment.S3: False}),
```
Comment (us) [inline tensorflow_datasets/image/cifar10_1.py]: Why s3 is false? I think you should add 2.0.0 version to s3 update.

File: tensorflow_datasets/image/cifar10_1.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      Cifar10_1Config(
+          name="v4",
+          description=(
+              "It is the first version of our dataset on which we tested any classifier. As mentioned above, this "
+              "makes the v4 dataset independent of the classifiers we evaluate. The numbers reported in the main "
+              "sections of our paper use this version of the dataset. It was built from the top 25 TinyImages "
+              "keywords for each class, which led to a slight class imbalance. The largest difference is that ships "
+              "make up only 8% of the test set instead of 10%. v4 contains 2,021 images."),
+          version=tfds.core.Version(
+              "1.0.0", experiments={tfds.core.Experiment.S3: False}),
```
Comment (pierrot0) [inline tensorflow_datasets/image/cifar10_1.py]: Remove `, experiments={tfds.core.Experiment.S3: False}`.
New datasets should use S3.

File: tensorflow_datasets/image/cifar10_1.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    image_path, label_path = dl_manager.download([
+        image_url,
+        label_url,
+    ])
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            num_shards=10,
```
Comment (pierrot0) [inline tensorflow_datasets/image/cifar10_1.py]: No longer necessary with S3.

**Comment by ChanchalKumarMaji** [comment]
> While generating url_checksum files, I get ERROR -
> 
> ```
> 0 examples [00:00, ? examples/s]Traceback (most recent call last):
>   File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
>     "__main__", mod_spec)
>   File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
>     exec(code, run_globals)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/scripts/download_and_prepare.py", line 210, in <module>
>     app.run(main)
>   File "/home/chanchal/dev_env/lib/python3.6/site-packages/absl/app.py", line 300, in run
>     _run_main(main, args)
>   File "/home/chanchal/dev_env/lib/python3.6/site-packages/absl/app.py", line 251, in _run_main
>     sys.exit(main(argv))
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/scripts/download_and_prepare.py", line 201, in main
>     download_and_prepare(builder_for_config)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/scripts/download_and_prepare.py", line 135, in download_and_prepare
>     download_config=dl_config,
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/api_utils.py", line 52, in disallow_positional_args_dec
>     return fn(*args, **kwargs)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 287, in download_and_prepare
>     download_config=download_config)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 948, in _download_and_prepare
>     max_examples_per_split=download_config.max_examples_per_split,
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 816, in _download_and_prepare
>     self._prepare_split(split_generator, **prepare_split_kwargs)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 965, in _prepare_split
>     return self._prepare_split_legacy(generator, split_info)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 955, in _prepare_split_legacy
>     self._file_format_adapter.write_from_generator(generator, output_files)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/file_format_adapter.py", line 126, in write_from_generator
>     _write_tfrecords_from_generator(wrapped, output_files, shuffle=True)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/file_format_adapter.py", line 234, in _write_tfrecords_from_generator
>     _round_robin_write(writers, generator)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/file_format_adapter.py", line 249, in _round_robin_write
>     generator, unit=" examples", leave=False)):
>   File "/home/chanchal/dev_env/lib/python3.6/site-packages/tqdm/_tqdm.py", line 1017, in __iter__
>     for obj in iterable:
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/file_format_adapter.py", line 124, in <genexpr>
>     wrapped = (self._serializer.serialize_example(example)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/dataset_builder.py", line 953, in <genexpr>
>     generator = (self.info.features.encode_example(ex) for ex in generator)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/features/features_dict.py", line 168, in encode_example
>     in utils.zip_dict(self._feature_dict, example_dict)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/features/features_dict.py", line 167, in <dictcomp>
>     for k, (feature, example_value)
>   File "/home/chanchal/Desktop/datasets-master/tensorflow_datasets/core/features/class_label_feature.py", line 143, in encode_example
>     if not -1 <= example_data < self._num_classes:
> ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
> ```
> 
> @Conchylicultor @us

Can u please check the error. I don't know why it is happening. @pierrot0

**Comment by pierrot0** [comment]
Can you debug and look at what are the values of `example_data` and `self._num_classes`?

**Pull request merged**